scapy/packet.py

```diff
@@ -1141,7 +1272,7 @@ def getlayer(self, cls, nb=1, _track=None, _subclass=None, **flt):
                     if fld is None:
                         return self
                     else:
-                        return self.getfieldval(fld)
+                        return self.getfieldval(fld)  # type: ignore
```

Comment:
I didn't understand the piece of code before this line (the for). Why do we use this for already? Can you get a field's value using `getlayer` ?!